Improve rdg storage format testing

e-mcginnis opened the pull request:
our existing tests only did:
1) take a graph without EntityTypeID structures
2) load it
3) store it in a new location, which stores the new EntityTypeID structures
4) load from the new location

added a new test that would have caught #488:
1) take a graph that already has EntityTypeID structures
2) load it
3) store it in a new location
4) load from the new location

Also cleaned up the test structure and added some sanity checks

unfortunately, the PR diff just shows the original file was deleted and a new file magically appeared instead of it properly showing a rename with code changes. 
Please see the individual commits instead.

arthurp reviewed (approved):
LGTM. The conda recipe reformat is just fine. Other changes look just fine, but I only spot checked.

e-mcginnis committed:
test load/store new format rdg

our existing tests only did:
1) take a graph without EntityTypeID structures
2) load it
3) store it in a new location, which stores the new EntityTypeID structures
4) load from the new location

added a new test that would have caught #488:
1) take a graph that already has EntityTypeID structures
2) load it
3) store it in a new location
4) load from the new location

Also cleaned up the test structure and added some sanity checks

e-mcginnis committed:
rename property graph rdg format conversion test

types-outside-properties.cpp -> property-file-graph-rdg-conversion.cpp

the existing name wasn't completely accurate in the first place
hopefully this one will be more clear

e-mcginnis committed:
format fix
